File: bin/gcoder_test.cc
#include "gcoder.cc"

#include <atomic>
#include <chrono>
#include <iostream>
#include <thread>

void MonitorThread( std::atomic<bool>& run_flag )
{
  unsigned counter = 0;

  while( run_flag ){
    std::cout << "Counter:" << counter  << std::endl;
    counter++;
    std::this_thread::sleep_for( std::chrono::milliseconds( 100 ) );
  }
}


int
main()
{
  GCoder gcoder;
  std::atomic<bool> runflag( true );
  std::thread mthread = std::thread( MonitorThread, std::ref( runflag ) );

  gcoder.InitPrinter( "/dev/ttyUSB0" );
  gcoder.MoveTo( 100, 100, 20 );
  std::this_thread::sleep_for( std::chrono::seconds( 1 ) );
  gcoder.MoveTo( 10, 10, 10  );
  std::this_thread::sleep_for( std::chrono::seconds( 1 ) );

  runflag = false;
  mthread.join();

  return 0;


}

File: bin/gcoder_test.py
import cmod.gcoder as gcoder
import threading
import time

run_flag = True


def monitor_thread():
  global run_flag
  counter = 0
  while run_flag:
    print("Counter at ", counter)
    counter = counter + 1
    time.sleep(0.1)
  return


if __name__ == "__main__":
  gc = gcoder.GCoder()
  mthread = threading.Thread(target=monitor_thread)
  mthread.start()
  try:
    gc.initprinter("/dev/ttyUSB0")
  except:
    pass
  try:
    gc.moveto(100, 100, 20, False)
  except:
    pass
  time.sleep(1)
  try:
    gc.moveto(10, 10, 1, False)
  except:
    pass
  time.sleep(1)

  run_flag = False
  mthread.join()
